packages/caliper-core/test/worker/rate-control/recordRate.js

```diff
@@ -63,42 +72,171 @@ describe('RecordRate controller', () => {
-                module: 'module.js'
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('Export Formats', () => {
+        it('should default outputFormat to TEXT if undefined', () => {
+            msgContent.rateControl.opts.outputFormat = undefined;
+            const testMessage = new TestMessage('test', [], msgContent);
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+            controller.outputFormat.should.equal('TEXT');
+        });
+
+
+        it('should set outputFormat to TEXT if invalid format is provided', () => {
+            msgContent.rateControl.opts.outputFormat = 'INVALID_FORMAT';
+            const testMessage = new TestMessage('test', [], msgContent);
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+
+            controller.outputFormat.should.equal('TEXT');
+        });
+
+        const formats = ['TEXT', 'BIN_BE', 'BIN_LE'];
+        const recordScenarios = [
+            {
+                description: 'with gaps (sparse records)',
+                records: { 1: 100, 3: 200, 7: 300 },
+                expectedLength: 8
             },
-            testRound: 0,
-            txDuration: 250,
-            totalWorkers: 2
-        };
-        const testMessage = new TestMessage('test', [], msgContent);
+            {
+                description: 'fully populated (sequential records)',
+                records: { 0: 50, 1: 100, 2: 150, 3: 200, 4: 250 },
+                expectedLength: 5
+            }
+        ];
+
+        formats.forEach(format => {
+            recordScenarios.forEach(scenario => {
+                it(`should export records to ${format} format ${scenario.description}`, async () => {
+                // Prepare message content with the specific output format
+                    const msgContentCopy = JSON.parse(JSON.stringify(msgContent));
+                    msgContentCopy.rateControl.opts.outputFormat = format;
+                    const testMessage = new TestMessage('test', [], msgContentCopy);
+                    const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+
+                    sinon.stub(controller.recordedRateController, 'end').resolves();
+                    controller.records = new Array(scenario.expectedLength);
+
+                    Object.keys(scenario.records).forEach(index => {
+                        controller.records[index] = scenario.records[index];
+                    });
+
+                    const fsWriteSyncStub = sandbox.stub(fs, 'writeFileSync');
+                    const fsAppendSyncStub = sandbox.stub(fs, 'appendFileSync');
+
+                    await controller.end();
+
+                    if (format === 'TEXT') {
+                        sinon.assert.calledOnce(fsWriteSyncStub);
+                        sinon.assert.callCount(fsAppendSyncStub, controller.records.length);
```

Comment:
this should assert to expectedLength and we should also assert that the records.length asserts to the expectedLength